dapps/shop/backend/utils/dns/clouddns.js

```diff
@@ -0,0 +1,177 @@
+ *
+ * @param {string} name of DNS record
+ * @returns {Zone}
+ */
+async function getZone(DNSName) {
+  const dns = getClient()
+  const [zones] = await dns.getZones({ maxResults: 50 })
+
+  for (let i=0; i<zones.length; i++) {
+    const dnsName = zones[i].metadata.dnsName
+    if (dnsName == DNSName) {
+      return zones[i]
+    }
+  }
+  return null
+}
+
+/**
+ * add a CNAME record
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Change.html
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Zone.html
+ *
+ * @param {Zone} zone that we're operating on
+ * @param {string} name of DNS record
+ * @param {string} target DNS name (e.g. whatever.com)
+ * @returns {Change}
+ */
+async function addCNAME(zone, name, target) {
+  const rec = zone.record('CNAME', {
+    name,
+    data: target,
+    ttl: DEFAULT_TTL
+  })
+  return await zone.addRecords(rec)
+}
+
+/**
+ * add a TXT record
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Zone.html
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Change.html
+ *
+ * @param {Zone} zone that we're operating on
+ * @param {string} name of DNS record
+ * @param {string} value of TXT record
+ * @returns {Change}
+ */
+async function addTXT(zone, name, txt) {
+  const rec = zone.record('TXT', {
+    name,
+    data: txt,
+    ttl: DEFAULT_TTL
+  })
+  return await zone.addRecords(rec)
+}
+
+/**
+ * add a DNSLink TXT record
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Zone.html
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Change.html
+ *
+ * @param {Zone} zone that we're operating on
+ * @param {string} name of DNS record
+ * @param {string} IPFS hash to point DNSLink towards
+ * @returns {Change}
+ */
+async function addDNSLink(zone, name, ipfsHash) {
+  return await addTXT(zone, `_dnslink.${name}`, `dnslink=/ipfs/${ipfsHash}`)
+}
+
+/**
+ * Set the necessary DNS records for a shop to a subdomain controlled by
+ * CloudDNS.
+ *
+ * Ref: https://googleapis.dev/nodejs/dns/latest/Change.html
+ *
+ * @param {object} args
+ * @param {string} args.credentials - The JSON Google service account
+ *  credentials
+ * @param {string} args.zone - The DNS zone we're adding records to
+ * @param {string} args.subdomain - The name of the record we're setting
+ * @param {string} args.ipfsGateway - The IFPS gateway to use for DNSLink
+ * @param {string} args.hash - The IPFS hash to use for DNSLink
+ * @returns {array} of Change
+ */
+async function setRecords({ credentials, zone, subdomain, ipfsGateway, hash }) {
+  const fqSubdomain = append(`${subdomain}.${zone}`, '.')
+  zone = append(zone, '.')
+  ipfsGateway = append(ipfsGateway, '.')
+
+  // Configure the client with given credentials
+  getClient(credentials)
+
+  const zoneObj = await getZone(zone)
+
+  if (!zoneObj || !(await zoneObj.exists())) {
+    console.error(`Zone ${zone} not found.`)
+    return
+  }
+
+  const records = zoneObj.getRecords({ maxResults: 250 })
+
+  if (await containsRecord(records, fqSubdomain)) {
```

Comment:
no need for await here. containsRecord is not asyn. also per prev comment, we can replace with ```if records.includes(fqSubdomain)```